packages/jest-cli/src/init/index.ts

```diff
@@ -122,6 +114,23 @@ export default async (
     return;
   }
 
+  // Determine if Jest should use JS or TS for the config file
+  const jestConfigFileExt = results.useTypescript
+    ? JEST_CONFIG_EXT_TS
+    : JEST_CONFIG_EXT_JS;
```

Comment:
```suggestion
  // Determine if Jest should use JS or TS for the config file
  const jestConfigFileExt = results.useTypescript
    ? JEST_CONFIG_EXT_TS
    : projectPackageJson.type === 'module' ? JEST_CONFIG_EXT_MJS : JEST_CONFIG_EXT_JS;
```

something like this (which prettier tweaks, but yeah)